client/request.go

```diff
@@ -35,11 +35,12 @@ type requestHandler func(operation []byte) <-chan []byte
 // makeRequestHandler constructs a requestHandler uisng the supplied
 // clientID, sequence number generator, authenticator, request buffer,
 // and number of tolerated faulty replicas.
-func makeRequestHandler(clientID uint32, seq sequenceGenerator, authen api.Authenticator, buf *requestbuffer.T, f uint32) requestHandler {
+func makeRequestHandler(clientID uint32, seq sequenceGenerator, authen api.Authenticator, collector ReplyCollector, buf *requestbuffer.T, f uint32) requestHandler {
 	submitter := makeRequestSubmitter(clientID, seq, authen, buf)
-	collector := makeReplyCollector(f, buf)
```

Comment:
Maybe we should rather pass `requestRemover` to `makeReplyCollector`.

Reply:
But that would force us to expose the RequestBuffer declared in `client/internal`, which is an object that is shared by requestRemover and other parts of the code that we do no want to expose. A developer that would want to supply its own ReplyCollector would have to be aware of this.

Reply:
If we leave it as is, the user of the proposed generic connectivity would have no control over this protocol-specific functionality. A generic code should not force MinBFT-specific `requestRemover`.

Reply:
Then I wonder if we should make RequestRemover external.

Reply:
If you envision how MinBFT-specific request remover may be generically useful.

Reply:
I'm thinking that maybe the better approach would be to enable developers to supply their own RequestHandler instead of a ReplyCollector. That way usage of a RequestRemover is at the discretion of the programmers, using whatever interface they want. It would also allow to specify how the request message is defined.

Reply:
Upon more exploration and reflection, I think there is no point in trying to make `client.New()` more generic. I could simply create my own implementation of the `Client` interface and re-use `ReplicaConnector`.